### PR TITLE
Rename `OP_*` constants to `OpCode*` constants 

### PR DESCRIPTION
**Pull request by seeforschauer**

### CODE REVIEW

**Review by w84thesun — APPROVED** [review]

File: internal/wire/msg_header.go [hunk 1]
```diff
@@ -29,16 +29,16 @@ import (
...
-	OP_UPDATE       = OpCode(2001) // OP_UPDATE
-	OP_INSERT       = OpCode(2002) // OP_INSERT
-	OP_GET_BY_OID   = OpCode(2003) // OP_GET_BY_OID
-	OP_QUERY        = OpCode(2004) // OP_QUERY
-	OP_GET_MORE     = OpCode(2005) // OP_GET_MORE
-	OP_DELETE       = OpCode(2006) // OP_DELETE
-	OP_KILL_CURSORS = OpCode(2007) // OP_KILL_CURSORS
-	OP_COMPRESSED   = OpCode(2012) // OP_COMPRESSED
-	OP_MSG          = OpCode(2013) // OP_MSG
+	OpCodeReply       = OpCode(1)    // OpCodeReply
```
Comment (AlekSi) [inline internal/wire/msg_header.go:32]: We should not change string representations – they follow protocol spec

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/wire/msg_header.go (hunk 1 above)
Comment (seeforschauer) [reply internal/wire/msg_header.go]: changed

File: internal/wire/msg_header.go [hunk 2]
```diff
@@ -29,16 +29,16 @@ import (
...
-	OP_QUERY        = OpCode(2004) // OP_QUERY
-	OP_GET_MORE     = OpCode(2005) // OP_GET_MORE
-	OP_DELETE       = OpCode(2006) // OP_DELETE
-	OP_KILL_CURSORS = OpCode(2007) // OP_KILL_CURSORS
-	OP_COMPRESSED   = OpCode(2012) // OP_COMPRESSED
-	OP_MSG          = OpCode(2013) // OP_MSG
+	OpCodeReply       = OpCode(1)    // OP_REPLY
+	OpCodeUpdate      = OpCode(2001) // OP_UPDATE
+	OpCodeInsert      = OpCode(2002) // OP_INSERT
+	OpCodeGetByOid    = OpCode(2003) // OP_GET_BY_OID
```
Comment (seeforschauer) [inline internal/wire/msg_header.go:35]: maybe here `OpCodeGetByOID`?

File: internal/wire/msg_header.go [hunk 3]
```diff
@@ -26,19 +26,45 @@ import (
...
-	OP_GET_MORE     = OpCode(2005) // OP_GET_MORE
-	OP_DELETE       = OpCode(2006) // OP_DELETE
-	OP_KILL_CURSORS = OpCode(2007) // OP_KILL_CURSORS
-	OP_COMPRESSED   = OpCode(2012) // OP_COMPRESSED
-	OP_MSG          = OpCode(2013) // OP_MSG
+	// OpCodeReply is sent by the database in response to an OP_QUERY or OP_GET_MORE message.
+	// Deprecated.
+	OpCodeReply = OpCode(1) // OP_REPLY
+
+	// OpCodeUpdate used to update a document in a collection.
```
Comment (rumyantseva) [inline internal/wire/msg_header.go:37]:
```suggestion
	// OpCodeUpdate is used to update a document in a collection.
```

For consistency with the below messages :)

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by rumyantseva — DISMISSED** [review]
I see that codecov complains, but its complaints are related to the previous changes, not to the current ones. So, these complaints are irrelevant to the current purpose of the PR.

LGTM.

File: internal/wire/msg_header.go [hunk 4]
```diff
@@ -26,19 +26,39 @@ import (
...
+	// OpCodeQuery is deprecated.
+	OpCodeQuery = OpCode(2004) // OP_QUERY
+
+	// OpCodeGetMore is deprecated.
+	OpCodeGetMore = OpCode(2005) // OP_GET_MORE
+
+	// OpCodeDelete is deprecated.
+	OpCodeDelete = OpCode(2006) // OP_DELETE
+
+	// OpCodeKillCursors is used to close an active cursor in the database.
```
Comment (AlekSi) [inline internal/wire/msg_header.go:54]:
```suggestion
	// OpCodeKillCursors is deprecated.
```

File: internal/wire/msg_header.go [hunk 5]
```diff
@@ -26,19 +26,39 @@ import (
...
-	OP_QUERY        = OpCode(2004) // OP_QUERY
-	OP_GET_MORE     = OpCode(2005) // OP_GET_MORE
-	OP_DELETE       = OpCode(2006) // OP_DELETE
-	OP_KILL_CURSORS = OpCode(2007) // OP_KILL_CURSORS
-	OP_COMPRESSED   = OpCode(2012) // OP_COMPRESSED
-	OP_MSG          = OpCode(2013) // OP_MSG
+	// OpCodeReply is deprecated.
+	OpCodeReply = OpCode(1) // OP_REPLY
+
+	// OpCodeUpdate is used to update a document in a collection.
```
Comment (AlekSi) [inline internal/wire/msg_header.go:36]:
```suggestion
	// OpCodeUpdate is deprecated.
```

File: internal/wire/msg_header.go [hunk 6]
```diff
@@ -26,19 +26,39 @@ import (
...
+	OpCodeUpdate = OpCode(2001) // OP_UPDATE
+
+	// OpCodeInsert is deprecated.
+	OpCodeInsert = OpCode(2002) // OP_INSERT
+
+	// OpCodeGetByOID reserved wire op code.
```
Comment (AlekSi) [inline internal/wire/msg_header.go:42]:
```suggestion
	// OpCodeGetByOID is deprecated.
```

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/wire/msg_header.go [hunk 7]
```diff
@@ -26,19 +26,39 @@ import (
...
-	OP_UPDATE       = OpCode(2001) // OP_UPDATE
-	OP_INSERT       = OpCode(2002) // OP_INSERT
-	OP_GET_BY_OID   = OpCode(2003) // OP_GET_BY_OID
-	OP_QUERY        = OpCode(2004) // OP_QUERY
-	OP_GET_MORE     = OpCode(2005) // OP_GET_MORE
-	OP_DELETE       = OpCode(2006) // OP_DELETE
-	OP_KILL_CURSORS = OpCode(2007) // OP_KILL_CURSORS
-	OP_COMPRESSED   = OpCode(2012) // OP_COMPRESSED
-	OP_MSG          = OpCode(2013) // OP_MSG
+	// OpCodeReply is deprecated.
```
Comment (AlekSi) [inline internal/wire/msg_header.go:33]: We use it, so it should have a better description

File: internal/wire/msg_header.go [hunk 8]
```diff
@@ -26,19 +26,39 @@ import (
...
+	// OpCodeUpdate is deprecated.
+	OpCodeUpdate = OpCode(2001) // OP_UPDATE
+
+	// OpCodeInsert is deprecated.
+	OpCodeInsert = OpCode(2002) // OP_INSERT
+
+	// OpCodeGetByOID is deprecated.
+	OpCodeGetByOID = OpCode(2003) // OP_GET_BY_OID
+
+	// OpCodeQuery is deprecated.
```
Comment (AlekSi) [inline internal/wire/msg_header.go:45]: We use it, so it should have a better description

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/wire/msg_header.go (hunk 7 above)
Comment (seeforschauer) [reply internal/wire/msg_header.go]: added.

File: internal/wire/msg_header.go (hunk 8 above)
Comment (seeforschauer) [reply internal/wire/msg_header.go]: added.

**Comment by seeforschauer** [comment]
added some comments, however, I acknowledge that it may differ how I understand the purpose from understanding of others.

**Review by w84thesun — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**